### PR TITLE
Fix #3495: Change Token to Bearer

### LINKED ISSUE #3495: Remote evaluation doesn't work
Running eval.ai with docker-compose and trying to connect with a remote worker does not work. It seems the remote worker script is a bit out of date with the current API. The [API documentation](https://eval.ai/api/docs/#operation/jobs_challenge_submission_list) helped out a lot in figuring out what the issue was.

These are the issues I had to fix to get the script to work:

* Rename `Token` to `Bearer` ([L329](https://github.com/Cloud-CV/EvalAI/blob/96c2b5e02bfbcceba6d1c672d20529a059fd6af8/scripts/workers/remote_submission_worker.py#L329)).
* Remove return statement ([L527](https://github.com/Cloud-CV/EvalAI/blob/96c2b5e02bfbcceba6d1c672d20529a059fd6af8/scripts/workers/remote_submission_worker.py#L527-L528)). I don't understand why this is there.
* Fix the `submission_data["result"]` data structure format. 
  The evaluation script is expected to return an object in the format of :
  ```
  [ 
    { "train-split" : { 
      "Metric" : 100 
    } } 
  ]
  ``` 
  but the API expects :
  ```
  [ 
    {
      "split": "train-split", 
      "show_to_participant" : True, 
      "accuracies" : { "Metric" : 100 } 
    }
  ]
  ```
  So a little bit of post-processing is needed. 

I will update #3496 to contain these fixes.

@Ram81: After I update the PR, can you review my changes?

### PR DESCRIPTION
**Pull request by rcannood**
Fixes #3495.

Running eval.ai with docker-compose and trying to connect with a remote worker does not work. It seems the remote worker script is a bit out of date with the current API. The [API documentation](https://eval.ai/api/docs/#operation/jobs_challenge_submission_list) helped out a lot in figuring out what the issue was.

These are the issues I had to fix to get the script to work:

* Rename `Token` to `Bearer` ([L329](https://github.com/Cloud-CV/EvalAI/blob/96c2b5e02bfbcceba6d1c672d20529a059fd6af8/scripts/workers/remote_submission_worker.py#L329)).
* Remove return statement ([L527](https://github.com/Cloud-CV/EvalAI/blob/96c2b5e02bfbcceba6d1c672d20529a059fd6af8/scripts/workers/remote_submission_worker.py#L527-L528)). I don't understand why this is there.
* Allow changing the protocol for contacting the django server from http to https by setting `export DJANGO_SERVER_PROTOCOL=https`.
* Fix the `submission_data["result"]` data structure format. 
  The evaluation script is expected to return an object in the format of :
  ```
  [ 
    { "train-split" : { 
      "Metric" : 100 
    } } 
  ]
  ``` 
  but the API expects :
  ```
  [ 
    {
      "split": "train-split", 
      "show_to_participant" : True, 
      "accuracies" : { "Metric" : 100 } 
    }
  ]
  ```
  So a little bit of post-processing is needed. I could have done the postprocessing in the evaluation script itself, but I thought it'd be better to keep the same output format for the evaluation script so that switching between remote evaluation or not can still happen fairly quickly without needing to change the output format of the evaluation script.

### CODE REVIEW

File: scripts/workers/remote_submission_worker.py [hunk 1]
```diff
@@ -25,11 +25,15 @@
 
 logger = logging.getLogger(__name__)
 
+logging.basicConfig(level=logging.INFO)
+
 AUTH_TOKEN = os.environ.get("AUTH_TOKEN")
 DJANGO_SERVER = os.environ.get("DJANGO_SERVER", "localhost")
 DJANGO_SERVER_PORT = os.environ.get("DJANGO_SERVER_PORT", "8000")
+DJANGO_SERVER_PROTOCOL = os.environ.get("DJANGO_SERVER_PROTOCOL", "http")
```
Comment (Ram81) [inline scripts/workers/remote_submission_worker.py:33]: @rcannood instead of taking protocol, port and server DNS explicitly we switched to passing the EvalAI endpoint as a single environment variable. So the value of endpoint would be `https://eval.ai` for prod and `http://localhost:8000` for dev environment. As we handle redirection of requests on our nginx reverse proxy we don't have to pass the port in the request url for production and staging servers.

File: scripts/workers/remote_submission_worker.py [hunk 2]
```diff
@@ -164,11 +168,12 @@ def create_dir_as_python_package(directory):
 
 
 def return_url_per_environment(url):
-    base_url = "http://{0}:{1}".format(DJANGO_SERVER, DJANGO_SERVER_PORT)
+    base_url = "{0}://{1}:{2}".format(DJANGO_SERVER_PROTOCOL, DJANGO_SERVER, DJANGO_SERVER_PORT)
```
Comment (Ram81) [inline scripts/workers/remote_submission_worker.py:171]: So this would become `base_url=EVALAI_API_SERVER` where `EVALAI_API_SERVER=os.environ.get("EVALAI_API_SERVER", "http://localhost:8000")`

**Review by Ram81 — CHANGES_REQUESTED** [review]
@rcannood rest of the changes in the PR looks good to me. If you are interested in switching to a single environment variable for EvalAI endpoint you can make the change and I can look at it again. If not, l'm happy to get a PR merged for this and you can use that if needed. 

Thanks for pointing out the issues with remote evaluation worker 🙂

File: scripts/workers/remote_submission_worker.py (hunk 2 above)
Comment (rcannood) [reply scripts/workers/remote_submission_worker.py:171]: That makes sense, I'll make the requested changes right away ^^

**Comment by rcannood** [comment]
Thanks for taking your time to review this PR ^^

I just pushed the requested changes to the patch-1 branch.

**Comment by rcannood** [comment]
Hello @Ram81 !

I made the requested changes, which wasn't a lot. Would you have some time to take a look at it? 

Thanks! :)
Robrecht

File: scripts/workers/remote_submission_worker.py [hunk 3]
```diff
@@ -563,9 +565,19 @@ def run_submission(
...
+                temp = {
+                    "split": key,
+                    "show_to_participant": True,
+                    "accuracies": value
+                }
+                new_res.append(temp)
+        submission_data["result"] = json.dumps(new_res)
         submission_data["metadata"] = json.dumps(
-            submission_output.get("submission_metadata")
+            submission_output.get("submission_result")
```
Comment (Ram81) [inline scripts/workers/remote_submission_worker.py:580]: @rcannood we allow hosts to add additional submission metadata if they want to. `result` and `submission_metadata` are two different files accessible to participants. Can we revert the changes on this line. Rest of the PR looks good to me.

**Review by Ram81 — CHANGES_REQUESTED** [review]
@rcannood just a minor change. Rest of the PR LGTM. Thanks for fixing the issues 🙂

**Review by rcannood — COMMENTED** [review]
Thanks @Ram81 ! I updated the code as requested.

Sorry for the delay, I was out of office for a while.

**Comment by rcannood** [comment]
Kind reminder for @Ram81. Is there anything blocking this PR from being merged? :bow:

**Comment by rcannood** [comment]
Closing pull request as you don't seem to be interested anymore.

Let me know if you want want me to fix the remote worker.

**Comment by Ram81** [comment]
Hi @rcannood, sorry about the delay. I had missed this PR. I am reopening this PR as it has a bunch of changes that would help set up remote evaluation challenges easily. Thanks for taking time to work on this PR.

**Comment by rcannood** [comment]
Closing this PR
